test/test_diffusion.py

```diff
@@ -624,6 +629,115 @@ def get_u2(x, t):
+        return numerical_flux_func(
+            kappa_tpair, u_tpair, grad_u_tpair, lengthscales_tpair, normal,
+            penalty_amount=penalty_amount)
+
+
+def test_orthotropic_diffusion(actx_factory):
+    """Checks that the diffusion operator can handle arrays for diffusivity."""
+    actx = actx_factory()
+    dim = 2
+    n = 4
+    order = 4
+
+    kappa0 = 2
+    kappa1 = 3
+
+    mesh = get_box_mesh(dim, -1, 1, n)
+    dcoll = create_discretization_collection(actx, mesh, order)
+    nodes = actx.thaw(dcoll.nodes())
+
+    zeros = actx.np.zeros_like(nodes[0])
+    kappa = make_obj_array([kappa0 + zeros, kappa1 + zeros])
+    u = 30.0*nodes[0] + 60.0*nodes[1]
+
+    boundaries = {
+        BoundaryDomainTag("-1"): NeumannDiffusionBoundary(-30.0),  # grad \cdot n
+        BoundaryDomainTag("+1"): NeumannDiffusionBoundary(+30.0),  # grad \cdot n
+        BoundaryDomainTag("-2"): NeumannDiffusionBoundary(-60.0),  # grad \cdot n
+        BoundaryDomainTag("+2"): NeumannDiffusionBoundary(+60.0),  # grad \cdot n
+    }
+
+    rhs, grad_u = diffusion_operator(dcoll, kappa=kappa, u=u,
+            boundaries=boundaries, return_grad_u=True,
+            gradient_numerical_flux_func=grad_facial_flux_weighted)
+
+    err_grad_x = actx.to_numpy(op.norm(dcoll, grad_u[0] - 30.0, np.inf))
+    err_grad_y = actx.to_numpy(op.norm(dcoll, grad_u[1] - 60.0, np.inf))
+    assert err_grad_x < 1.e-9
+    assert err_grad_y < 1.e-9
+
+    diff_flux = diffusion_flux(kappa, grad_u)
+    flux_x = -(kappa0 + zeros)*grad_u[0]
+    flux_y = -(kappa1 + zeros)*grad_u[1]
+    err_flux_x = actx.to_numpy(op.norm(dcoll, diff_flux[0] - flux_x, np.inf))
+    err_flux_y = actx.to_numpy(op.norm(dcoll, diff_flux[1] - flux_y, np.inf))
+    assert err_flux_x < 1.e-9
+    assert err_flux_y < 1.e-9
+
+    err_rhs = actx.to_numpy(op.norm(dcoll, rhs, np.inf))
+    assert err_rhs < 1.e-8
+
+    boundaries = {
+        BoundaryDomainTag("-1"): DummyDiffusionBoundary(),
+        BoundaryDomainTag("+1"): DummyDiffusionBoundary(),
+        BoundaryDomainTag("-2"): DummyDiffusionBoundary(),
+        BoundaryDomainTag("+2"): DummyDiffusionBoundary(),
+    }
+
+    rhs, grad_u = diffusion_operator(dcoll, kappa=kappa, u=u,
+            boundaries=boundaries, return_grad_u=True,
+            gradient_numerical_flux_func=grad_facial_flux_weighted)
+
+    err_grad_x = actx.to_numpy(op.norm(dcoll, grad_u[0] - 30.0, np.inf))
+    err_grad_y = actx.to_numpy(op.norm(dcoll, grad_u[1] - 60.0, np.inf))
+    assert err_grad_x < 1.e-9
+    assert err_grad_y < 1.e-9
+
+    diff_flux = diffusion_flux(kappa, grad_u)
+    flux_x = -(kappa0 + zeros)*grad_u[0]
+    flux_y = -(kappa1 + zeros)*grad_u[1]
+    err_flux_x = actx.to_numpy(op.norm(dcoll, diff_flux[0] - flux_x, np.inf))
+    err_flux_y = actx.to_numpy(op.norm(dcoll, diff_flux[1] - flux_y, np.inf))
+    assert err_flux_x < 1.e-9
+    assert err_flux_y < 1.e-9
+
+    err_rhs = actx.to_numpy(op.norm(dcoll, rhs, np.inf))
+    assert err_rhs < 1.e-8
+
+
```

Comment:
What is this case testing vs. the other one with the `NeumannDiffusionBoundary`s?

Reply:
I just wanted to make sure it was working with both BCs (Dirichlet-like and Neumann)

Reply:
How about something like
```diff
     err_rhs = actx.to_numpy(op.norm(dcoll, rhs, np.inf))
     assert err_rhs < 1.e-8
 
+    def make_dirichlet_bc(btag):
+        bdry_u = op.project(dcoll, "vol", BoundaryDomainTag(btag), u)
+        return DirichletDiffusionBoundary(bdry_u)
+
     boundaries = {
-        BoundaryDomainTag("-1"): DummyDiffusionBoundary(),
-        BoundaryDomainTag("+1"): DummyDiffusionBoundary(),
-        BoundaryDomainTag("-2"): DummyDiffusionBoundary(),
-        BoundaryDomainTag("+2"): DummyDiffusionBoundary(),
+        BoundaryDomainTag("-1"): make_dirichlet_bc("-1"),
+        BoundaryDomainTag("+1"): make_dirichlet_bc("+1"),
+        BoundaryDomainTag("-2"): make_dirichlet_bc("-2"),
+        BoundaryDomainTag("+2"): make_dirichlet_bc("+2"),
     }
```
instead? I don't know if it's well posed with all dummy BCs.

Reply:
Done